internal/common/server_status.go

```diff
@@ -48,31 +57,32 @@ func GetUnixSocketPath() (string, error) {
 	return statusJson.UnixSocketPath, nil
 }
 
-func WriteAndHandleStatusFile() error {
-	Lock(PID_SOCK_LOCK)
-	Unlock(PID_SOCK_LOCK)
+func WriteAndHandleStatusFile(initiator ServerInitiator) error {
+	Lock(SERVER_STATUS_LOCK)
+	Unlock(SERVER_STATUS_LOCK)
 
 	statusPath := GetStatusPath()
 	pid := os.Getpid()
-	var statusJson StatusInfo
 
-	ok, err := checkAndHandleStalePid()
+	statusInfo, err := CheckAndHandleServerStatusFile()
 	if err != nil {
 		return err
 	}
 
-	if !ok {
-		return fmt.Errorf("Failed to create the status file '%s'.", statusPath)
+	if statusInfo != nil {
+		return fmt.Errorf("Failed to create the status file '%s' since another server is running.", statusPath)
```

Comment:
This comment is inaccurate.
It didn't fail to create another status file, it refused to do so.
Be more direct here, like:
"Cannot start server, another server is running (pid %d).".